src/raft.rs

```diff
@@ -1052,37 +1051,28 @@ impl<T: Storage> Raft<T> {
             (MessageType::MsgRequestVote, self.term)
         };
         let self_id = self.id;
-        let acceptance = true;
-        info!(
-            self.logger,
-            "{id} received message from {from}",
-            id = self.id,
-            from = self_id;
-            "msg" => ?vote_msg,
-            "term" => self.term
-        );
-        self.register_vote(self_id, acceptance);
-        if let Some(true) = self.check_votes() {
+        if VoteResult::Won == self.poll(self_id, vote_msg, true) {
             // We won the election after voting for ourselves (which must mean that
             // this is a single-node cluster).
             return;
         }
 
+        let mut voters = [0; 7];
+        let mut voter_cnt = 0;
+
         // Only send vote request to voters.
-        for id in self.prs.voter_ids() {
-            if *id == self_id {
+        for id in self.prs.voter_ids().iter() {
+            if id == self_id {
                 continue;
             }
-            info!(
-                self.logger,
-                "[logterm: {log_term}, index: {log_index}] sent request to {id}",
-                log_term = self.raft_log.last_term(),
-                log_index = self.raft_log.last_index(),
-                id = id;
-                "term" => self.term,
-                "msg" => ?vote_msg,
-            );
-            let mut m = new_message(*id, vote_msg, None);
+
+            if voter_cnt == 7 {
```

Comment:
Maybe use `voters.len()` instead

Reply:
voters is an array, its length is always 7.

Reply:
I think it could be better to use `voter_cnt == voters.len()` to replace the magic number